Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -192,8 +192,12 @@ protected override async Task OnAfterRenderAsync( bool firstRender )
                 } );
 
                 if ( ManualReadMode )
+                {
                     await HandleReadData( CancellationToken.None );
 
+                    return;

```

Comment:
The return is needed here as it was not working well without it. I remember there was a problem a few months ago.

Please test once again to confirm I didn't break it.

Reply:
That's hella strange. Are you sure the problem had to actually do with that? It's fine with me as long as it works. I'm testing and looks good to me.

Reply:
I can't find the actual merge that fixed it. I'm really not sure right now.. 

Just in case let's leave it for now. All it does will skip the addition call to StateHasChanged after HandleReadData finishes.

Reply:
Maybe if before StateHasChanged was called without InvokeAsync?
Anyway, I was just figuring out it may not be fine, it skips the base.OnAfterRenderAsync which might have logic... and I was just doing the AfterRenderQueue on the base... hmmm